src/pykeen/metrics/ranking.py

```diff
@@ -74,12 +79,107 @@ def expected_value(
         Compute expected metric value.
 
         Prefers analytical solution, if available, but falls back to numeric estimation via summation,
-        cf. `numeric_expected_value`.
+        cf. :func:`numeric_expected_value`.
         """
         if num_samples is None:
             raise ValueError("Numeric estimation requires to specify a number of samples.")
         return self.numeric_expected_value(num_candidates=num_candidates, num_samples=num_samples)
 
+    def numeric_variance(
+        self,
+        num_candidates: np.ndarray,
+        num_samples: int,
+    ) -> float:
+        """Compute variance by summation."""
+        num_candidates = np.asarray(num_candidates)
+        generator = np.random.default_rng()
+        ranks = generator.integers(low=1, high=num_candidates + 1, size=(num_samples, *num_candidates.shape))
```

Comment:
can we re-use `_yield_expected_values`?

Reply:
I did but the results weren't right so I switched back to this